recipes/dbus/all/conanfile.py

```diff
@@ -0,0 +1,115 @@
+from conans import CMake, ConanFile, tools
+import os
+
+
+class DbusConan(ConanFile):
+    name = "dbus"
+    license = "AFL-2.1"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.freedesktop.org/wiki/Software/dbus"
+    description = "D-Bus is a simple system for interprocess communication and coordination."
+    topics = ("conan", "dbus")
+    settings = "os", "compiler", "build_type", "arch"
+
+    options = {
+            "with_x11": [True, False], 
+            "with_glib": [True, False],
+            "disable_assert": [True, False],
+            "disable_checks": [True, False],
+            "enable_ansi": [True, False],
+            "enable_containers": [True, False],
+            "enable_stats": [True, False],
+            "enable_verbose_mode": [True, False],
+            "gcov_enabled": [True, False],
+            "install_system_libs": [True, False],
+            "use_output_debug_string": [True, False]}
+
+    default_options = {
+            "with_x11": False,
+            "with_glib": False,
+            "disable_assert": False,
+            "disable_checks": False,
+            "enable_ansi": False,
+            "enable_containers": False,
+            "enable_stats": False,
+            "enable_verbose_mode": True,
+            "gcov_enabled": False,
+            "install_system_libs": False,
+            "use_output_debug_string": False}
+
+    generators = "cmake_find_package", "cmake_paths"
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _cmake = None
+
+    requires = (
+            "expat/2.2.9"
+            )
+
+    def source(self):
+
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+        dbus_cmake = tools.os.path.join(self._source_subfolder, "cmake", "CMakeLists.txt")
+        dbus_cmake_tools = tools.os.path.join(self._source_subfolder, "cmake", 
+                "tools", "CMakeLists.txt")
+
+        if self.options.with_glib:
+            tools.replace_in_file(dbus_cmake, "GLib2", "glib")
+            tools.replace_in_file(dbus_cmake, "GLIB2", "GLIB")
+
+        if self.options.with_x11:
+            tools.replace_in_file(dbus_cmake, "X11", "libx11")
+            tools.replace_in_file(dbus_cmake_tools, "X11", "libx11")
+
+
+    def requirements(self):
+        if self.options.with_glib:
+            self.requires("glib/2.65.0")
+
+        if self.options.with_x11:
+            self.requires("xorg/system")
+
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+
+            self._cmake.definitions["DBUS_BUILD_TESTS"] = "OFF"
+            self._cmake.definitions["DBUS_ENABLE_DOXYGEN_DOCS"] = "OFF"
+            self._cmake.definitions["DBUS_ENABLE_XML_DOCS"] = "OFF"
```

Comment:
Does not matter much, but looks nicer.
```suggestion
            self._cmake.definitions["DBUS_BUILD_TESTS"] = False
            self._cmake.definitions["DBUS_ENABLE_DOXYGEN_DOCS"] = False
            self._cmake.definitions["DBUS_ENABLE_XML_DOCS"] = False
```